packages/react-router-dom/modules/BrowserRouter.js

```diff
@@ -18,6 +19,12 @@ class BrowserRouter extends React.Component {
   history = createHistory(this.props)
 
   render() {
+    warning(
+      !this.props.history,
+      '`<BrowserRouter history={...}` prop has been ignored. For custom history, ' +
+      'make sure to `import { Router }` and not `import { ... as Router }`.'
```

Comment:
I thought of it as a general warning to avoid using any of the Browser/Hash/Memory/Static Router with history, do you think the ellipsis makes the message unclear?